core/src/main/java/org/apache/iceberg/rest/ExponentialHttpRequestRetryStrategy.java

```diff
@@ -117,7 +140,15 @@ public boolean retryRequest(
 
   @Override
   public boolean retryRequest(HttpResponse response, int execCount, HttpContext context) {
-    return execCount <= maxRetries && retriableCodes.contains(response.getCode());
+    HttpRequest request =
+        context instanceof HttpCoreContext ? ((HttpCoreContext) context).getRequest() : null;
```

Comment:
I think by default the context object is null, we reverted setting the context :) 

as we call this API  from HTTPClient#execute

```
    @Override
    public CloseableHttpResponse execute(
            final ClassicHttpRequest request) throws IOException {
        return doExecute(determineTarget(request), request, null);
    }

```

we need to add this before the execute and pass :) !
HttpContext context = HttpClientContext.create();

Reply:
ah good catch, I'll go back and change that